test/test_irods_http_api.py

```diff
@@ -1648,6 +1650,177 @@ def test_parallel_writes(self):
         self.assertEqual(r.status_code, 200)
         self.assertEqual(r.json()['irods_response']['status_code'], 0)
 
+    def test_parallel_writes_with_data_exceeding_internal_write_threshold(self):
+        # This test assumes the HTTP API is configured to use a value smaller than
+        # 96kb for "/irods_client/buffer_size_in_bytes_for_write_operations". This is
+        # important because values greater than or equal to 96kb will result in the
+        # background thread pool only being used one time. This test is designed to
+        # force the HTTP API to schedule multiple writes across the thread pool.
+        #
+        # NOTE: This test DOES NOT and CANNOT confirm the background thread pool is
+        # being used because it does not have visibility into the server's internal
+        # mechanisms. Our only means of verification is by careful review of the log
+        # messages. The server is expected to output messages about multiple writes
+        # to the iRODS server. The thread ID should change from one write operation to
+        # the next.
+
+        headers = {'Authorization': f'Bearer {self.rodsuser_bearer_token}'}
+        data_object = f'/{self.zone_name}/home/{self.rodsuser_username}/incremental_writes_0.txt'
+
+        # Generate 96kb string.
+        # This is the data we'll upload to the HTTP API.
+        data96kb = ('A' * 32768) + ('B' * 32768) + ('C' * 32768)
+
+        # Calculate a checksum for data.
+        # The checksum will be used to confirm the data was uploaded correctly.
+        checksum = base64.b64encode(hashlib.sha256(data96kb.encode('utf-8')).digest()).decode('utf-8')
+        self.logger.debug(f'checksum = [{checksum}]')
+
+        # Indicates whether the parallel_write_shutdown operation needs to be called.
+        # This is primarily used for clean up when errors occur.
+        invoke_parallel_write_shutdown = True
```

Comment:
Should we only set this to `True` after `parallel_write_init` has completed successfully? I think `parallel_write_handle` will not be defined should either of the assumptions following `parallel_write_init` fail, which would cause problems in the cleanup. Until `parallel_write_init` succeeds, I think it should be set to `False`.

Also, I would argue the word "primarily" can be removed from the comment as it doesn't appear to be used anywhere else besides the cleanup.

Reply:
Done